test/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/SelectWhoIsMakingTheClaimControllerSpec.scala

```diff
@@ -105,201 +111,166 @@ class SelectWhoIsMakingTheClaimControllerSpec
 
     "display the page" when {
 
-      "the user has not answered this question before" in {
-        def performAction(): Future[Result] = controller.selectDeclarantType()(FakeRequest())
-
-        val answers = IncompleteDeclarantTypeAnswer.empty
+      def performAction(journeyBindable: JourneyBindable): Future[Result] =
+        controller.selectDeclarantType(journeyBindable)(FakeRequest())
 
-        val draftC285Claim                = sessionWithClaimState(Some(answers))._3
+      "the user has not answered this question before" in forAll(testCases) { (numberOfClaims, journeyBindable) =>
+        val draftC285Claim                = sessionWithClaimState(None, Some(numberOfClaims))._3
           .copy(movementReferenceNumber = sampleEntryNumberAnswer())
-        val (session, fillingOutClaim, _) = sessionWithClaimState(Some(answers))
-
-        val updatedJourney = fillingOutClaim.copy(draftClaim = draftC285Claim)
+        val (session, fillingOutClaim, _) = sessionWithClaimState(None, Some(numberOfClaims))
+        val updatedJourney                = fillingOutClaim.copy(draftClaim = draftC285Claim)
 
         inSequence {
           mockAuthWithNoRetrievals()
           mockGetSession(session.copy(journeyStatus = Some(updatedJourney)))
         }
 
         checkPageIsDisplayed(
-          performAction(),
+          performAction(journeyBindable),
           messageFromMessageKey("select-who-is-making-the-claim.title")
         )
-      }
-
-      "the user has answered this question before" in {
-        def performAction(): Future[Result] = controller.selectDeclarantType()(FakeRequest())
 
-        val declarantType = DeclarantType.Importer
-
-        val answers = CompleteDeclarantTypeAnswer(declarantType)
+      }
 
-        val draftC285Claim                = sessionWithClaimState(Some(answers))._3
+      "the user has answered this question before" in forAll(testCases) { (numberOfClaims, journeyBindable) =>
+        val declarantType                 = DeclarantTypeAnswer.Importer
+        val answers                       = declarantType
+        val draftC285Claim                = sessionWithClaimState(Some(answers), Some(numberOfClaims))._3
           .copy(movementReferenceNumber = sampleEntryNumberAnswer())
-        val (session, fillingOutClaim, _) = sessionWithClaimState(Some(answers))
-
-        val updatedJourney = fillingOutClaim.copy(draftClaim = draftC285Claim)
+        val (session, fillingOutClaim, _) = sessionWithClaimState(Some(answers), Some(numberOfClaims))
+        val updatedJourney                = fillingOutClaim.copy(draftClaim = draftC285Claim)
 
         inSequence {
           mockAuthWithNoRetrievals()
           mockGetSession(session.copy(journeyStatus = Some(updatedJourney)))
         }
 
         checkPageIsDisplayed(
-          performAction(),
+          performAction(journeyBindable),
           messageFromMessageKey("select-who-is-making-the-claim.title")
```

Comment:
Can you define "select-who-is-making-the-claim" as key in the controller and reference this variable in the tests?

Reply:
done